### PR TITLE
BAMPFA-584: Media customizations 

### PR DESCRIPTION
**Pull request by cesarvh**

### CODE REVIEW

File: src/plugins/recordTypes/index.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+import media from './media';
 import organization from './organization';
 import person from './person';
 
 export default [
   concept,
+  media,
   organization,
   person,
+  concept,
```
Comment (ray-lee) [inline src/plugins/recordTypes/index.js]: Extra concept.

File: src/plugins/recordTypes/media/forms/default.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  const {
+    extensions,
+  } = pluginContext.config;
+
+  return (
+    <Field name="document">
+      <Panel name="media" collapsible>
+        <Panel name="file" collapsible>
+          <Field name="page" subpath="ns2:media_cinefiles" />
```
Comment (ray-lee) [inline src/plugins/recordTypes/media/forms/default.jsx]: I'd like to move page up, out of the File Information panel, so it's the first field in Media Handling Information. Also try putting it in a row with an empty column at the end, so it only takes up half the width. It looks too wide to me.

**Review by ray-lee — COMMENTED** [review]
Hi Cesar,
A couple tweaks needed, see comments. Also I see a few more changes that need to be made:

Change label of title field to "File name" (appears on advanced search form).

Stuff from config/media.json in the 4.1 UI layer config:

Make page not cloneable (was fieldsToIgnore in 4.1, now cloneable: false in fields.js)

Make record title include page number 

In 4.1 media.json there's an "originalMediaDimensions" setting. Can you try to figure out what that does? I don't remember. Maybe start with a blame in git.

Thanks!

**Comment by cesarvh** [comment]
Hi Ray,

I went ahead and made the changes but for the media `title` field, once I added the `page number` field to the computation, the order is correct at first, but if the record is saved, then the page number is changed the order of the elements is swapped to `{file name} {page number}` instead of `{page number} {file name}`. I'm guessing this is because React only renders what is needed, but I'm not entirely sure.

**Comment by cesarvh** [comment]
Hi Ray,

I finished the customizations and looked into the originalMediaDimensions. Through blame, I was able to find the commit that appears to be the oldest regarding this. 
```
commit 9f7487b46b27bcde37a4d7096c8e82e66bf3854f
Author: Alexey Novak <abovebluesky@gmail.com>

    CSPACE-5427: Fixing click on the media image in the record for Media Records

diff --git a/src/main/webapp/defaults/js/RecordEditor.js b/src/main/webapp/defaults/js/RecordEditor.js
--- a/src/main/webapp/defaults/js/RecordEditor.js
+++ b/src/main/webapp/defaults/js/RecordEditor.js
@@ -1518,9 +1518,9 @@

-    cspace.recordEditor.navigateToFullImage = function (model, originalMediaDimensions, parentBundle) {
+    cspace.recordEditor.recordRenderer.navigateToFullImage = function (model, originalMediaDimensions, parentBundle) {
         window.open(model.fields.blobs[0].imgOrig, "_blank", parentBundle.resolve("media-originalMediaOptions", [
             originalMediaDimensions.height,
             originalMediaDimensions.width,
             "yes"
         ]));
     };
```
I was also able to find this regarding its usage:
```
// Function to open an associated image with the record in the separate window
that.getOriginalImage = function () {
    var src = that.getMedia(that.model.currentMedia);
    window.open(src, "_blank", that.options.parentBundle.resolve("media-originalMediaOptions", ["600", "800", "yes"]));
    };
};
```
Though I'm not entirely sure, it appears to have to do something with when an image is clicked in a media record and it gets opened in another tab/window. I looked at the dimensions of when this happened, and they didn't seem to match up to 600x800 pixels. Though, according to the documentation, the `window.open()` function has the parameters `URL, name, specs, replace` which would match up to `URL=src, name="_blank", specs=that.options.parentBundle.resolve("media-originalMediaOptions", ["600", "800", "yes"])`. So this seems to specify the height of the pop-up window.

Also, the documentation states that it takes a string of named parameters, such as `window.open("", "", "width=200,height=100");`. I played around with some of W3 School's examples [here](https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_win_open2) and found that if the parameters are named, then the window opens with the correct dimensions, but if I pass in an array of options instead, the window does not open with the right sizes, which might explain why the dimensions didn't match up when I tried it on the media record.

**Comment by ray-lee** [comment]
Ok, I'm not going to worry about originalMediaDimensions. At any rate it's not something that was customized in cinefiles.

**Comment by ray-lee** [comment]
I think there's some confusion about the title. The media_common:title field is computed from the blob filename, and it should just contain the filename. You had that right. That's a denormalization of the blob filename into the media record, so people can search for media records by filename.

There is also a configurable title function for the record (title.js) that is used to determine what fields should be shown in the title bar. By default it shows the identificationNumber field and the title field. That is what needs to be changed. We hid the identificationNumber field, so it needs to be replaced with the page number.

**Comment by cesarvh** [comment]
Ah, that makes sense. I think I just realized that earlier this weekend while configuring title.js for the collection object record. I'll fix this today. Thanks!

**Pull request merged**